codility_training/lessons.lesson10.PrimeAndCompositeNumbers.Peaks/sjjyy.java

```diff
@@ -0,0 +1,47 @@
+package lessons.lesson10.PrimeAndCompositeNumbers.Peaks;
+
+import java.util.ArrayList;
+
+public class sjjyy
+{
+    public static int solution(int[] A)
+    {
+        int len = A.length;
+        ArrayList<Integer> peaks = new ArrayList<>();
+
+        for(int i = 1 ; i < len-1 ; i++)
+        {
+            if(A[i-1] < A[i] && A[i+1] < A[i])
+                peaks.add(i);
+        }
+
+        if(peaks.size() <= 1)
+            return peaks.size();
+
+        for(int num = len ; num >= 1 ; num--)
+        {
+            if (len % num == 0)
+            {
+                int size = len / num;
+                int block = 0;
+
+                for(int j : peaks)
+                {
+                    if(j / size == block)
+                        block++;
+                }
+
+                if(block == num)
+                    return num;
+            }
```

Comment:
```suggestion
            if (len % num != 0) continue;
            
            int size = len / num;
            int block = peaks.reduce(0, (n, j) => n + (j / size == n ? 1 : 0));
            
            if(block == num) return num;
```
reduce를 사용하면 더 간단하게 표현할 수 있습니다!